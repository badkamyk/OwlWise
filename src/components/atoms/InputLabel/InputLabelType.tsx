export type InputLabelType = {
	text: string;
	htmlFor: string;
}