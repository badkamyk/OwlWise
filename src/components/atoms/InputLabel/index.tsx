import { InputLabelType } from "./InputLabelType";

export default function InputLabel({ text, htmlFor }: InputLabelType) {
	return (
		<label
			htmlFor={htmlFor}
			className="mb-2 block text-sm font-medium text-gray-900 dark:text-gray-300"
		>
			{text}
		</label>
	);
}
