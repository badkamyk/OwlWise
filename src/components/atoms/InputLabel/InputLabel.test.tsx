import { render, screen } from "@testing-library/react";
import { expect, it } from "vitest";
import InputLabel from "./index";

it("InputLabel with props", () => {
	const { container } = render(<InputLabel text="Name" htmlFor="name" />);
	expect(screen.getByText("Name")).toBeTruthy();
	expect(container).toMatchSnapshot();
});
