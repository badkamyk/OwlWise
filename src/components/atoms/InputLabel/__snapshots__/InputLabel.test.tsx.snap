// Vitest Snapshot v1

exports[`InputLabel with props 1`] = `
<div>
  <label
    class="mb-2 block text-sm font-medium text-gray-900 dark:text-gray-300"
    for="name"
  >
    Name
  </label>
</div>
`;
